File: 2.Implement/Delivery_Server/src/main/java/com/delivery/app/Delivery/data/dto/request/user/SignUpDto.java
package com.delivery.app.Delivery.data.dto.request.user;

import jakarta.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class SignUpDto {
    private String name;
    private String nickname;
    private String loginId;
    private String loginPwd;
    private String phoneNumber;
    private String gender;
    private String address;

    public boolean hasNull() {
        return     name == null
                || nickname == null
                || loginId == null
                || loginPwd == null
                || phoneNumber == null
                || gender == null
                || address == null;
    }
}

File: 2.Implement/DeliveryApp/app/src/main/java/com/project/deliveryapp/retrofit/dto/request/user/SignUpDto.kt
package com.project.deliveryapp.retrofit.dto.request.user

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class SignUpDto(
    val name: String,
    val nickname: String,
    val loginId: String,
    val loginPwd: String,
    val phoneNumber: String,
    val gender: String,
    val address: String,
): Parcelable {
    fun isOkay(): Boolean {
        return     name.isNotBlank()
                && nickname.isNotBlank()
                && loginId.isNotBlank()
                && loginPwd.isNotBlank()
                && phoneNumber.isNotBlank()
                && gender.isNotBlank()
                && address.isNotBlank()
    }
}
